File: src/test/java/com/github/vbauer/avconv4java/model/TypeTest.java
package com.github.vbauer.avconv4java.model;

import com.github.vbauer.avconv4java.common.TestUtils;
import org.testng.Assert;
import org.testng.annotations.DataProvider;
import org.testng.annotations.Factory;
import org.testng.annotations.Test;

/**
 * @author Vladislav Bauer
 */

public class TypeTest {

    @DataProvider
    public Object[][] data() {
        return new Object[][] {
            { AVCodecFlagType.Constants.class },
            { AVDebugInfoType.Constants.class },
            { AVErrorDetectionType.Constants.class },
            { AVAudioCodecType.Constants.class },
            { AVFileFormatType.Constants.class },
            { AVFormatDebugInfoType.Constants.class },
            { AVFormatFlagType.Constants.class },
            { AVHardwareAccelerationType.Constants.class },
            { AVLogLevelType.Constants.class },
            { AVMotionEstimationType.Constants.class },
            { AVMovFlagsType.Constants.class },
            { AVStreamType.Constants.class },
            { AVStrictType.Constants.class },
            { AVTargetFileType.Constants.class },
            { AVVideoCodecType.Constants.class },
            { AVVideoSizeType.Constants.class },
            { AVVideoSyncType.Constants.class }
        };
    }

    @Factory(dataProvider = "data")
    public Object[] createTest(final Class<?> utilClass) {
        return new Object[] {
            new TypeConstructorTest(utilClass)
        };
    }


    /**
     * @author Vladislav Bauer
     */
    public static class TypeConstructorTest extends Assert {

        private final Class<?> utilClass;

        public TypeConstructorTest(final Class<?> utilClass) {
            this.utilClass = utilClass;
        }

        @Test(expectedExceptions = UnsupportedOperationException.class)
        public void testConstructorContract() throws Throwable {
            TestUtils.checkUtilClassConstructor(utilClass);
        }
    }

}

File: src/test/kotlin/com/github/vbauer/avconv4java/type/TypeTest.kt
package com.github.vbauer.avconv4java.type

import com.github.vbauer.avconv4java.common.TestUtils
import org.testng.Assert
import org.testng.annotations.DataProvider
import org.testng.annotations.Factory
import org.testng.annotations.Test

/**
 * @author Vladislav Bauer
 */

class TypeTest {

    @DataProvider
    fun data(): Array<Array<out Class<out Any>>> {
        return arrayOf(
                arrayOf(AVCodecFlagType.Constants::class.java),
                arrayOf(AVDebugInfoType.Constants::class.java),
                arrayOf(AVErrorDetectionType.Constants::class.java),
                arrayOf(AVAudioCodecType.Constants::class.java),
                arrayOf(AVFileFormatType.Constants::class.java),
                arrayOf(AVFormatDebugInfoType.Constants::class.java),
                arrayOf(AVFormatFlagType.Constants::class.java),
                arrayOf(AVHardwareAccelerationType.Constants::class.java),
                arrayOf(AVLogLevelType.Constants::class.java),
                arrayOf(AVMotionEstimationType.Constants::class.java),
                arrayOf(AVMovFlagsType.Constants::class.java),
                arrayOf(AVStreamType.Constants::class.java),
                arrayOf(AVStrictType.Constants::class.java),
                arrayOf(AVTargetFileType.Constants::class.java),
                arrayOf(AVVideoCodecType.Constants::class.java),
                arrayOf(AVVideoSizeType.Constants::class.java),
                arrayOf(AVVideoSyncType.Constants::class.java)
        )
    }

    @Factory(dataProvider = "data")
    fun createTest(utilClass: Class<*>): Array<Any> {
        return arrayOf(TypeConstructorTest(utilClass))
    }


    /**
     * @author Vladislav Bauer
     */
    class TypeConstructorTest internal constructor(private val utilClass: Class<*>) : Assert() {

        @Test
        fun testConstructorContract() {
            TestUtils.checkUtilClassConstructor(utilClass)
        }
    }

}
